### PR TITLE
haskell-modules: Fixes build for beam-* and postgresql-simple-migration

### PR DESCRIPTION
**Pull request by zzantares**
Following packages now are able to build:
  - `beam-core`
  - `beam-migrate`
  - `beam-postgres` (unsure about removing existing override)
  - `postgresql-simple-migration`

###### Motivation for this change

Beam packages are marked as broken due version bounds on `aeson` dependency not satisfied, by applying `doJailbreak` they build and work fine.

###### Things done

On `NixOS`, the following now works:

```
$ nix-build --no-out-link -A haskellPackages.beam-core --arg config '{ allowBroken = true; }'
$ nix-build --no-out-link -A haskellPackages.beam-migrate --arg config '{ allowBroken = true; }'
$ nix-build --no-out-link -A haskellPackages.beam-postgres --arg config '{ allowBroken = true; }'
$ nix-build --no-out-link -A haskellPackages.postgresql-simple-migration --arg config '{ allowBroken = true; }'
```

### CODE REVIEW

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1027,10 +1027,16 @@ self: super: {
...
-  beam-postgres = overrideCabal super.beam-postgres (drv: {
+  # Aeson dependency version bound not satifised but it builds fine without it
+  beam-core = doJailbreak super.beam-core;
+
+  # Aeson dependency version bound not satifised but it builds fine without it
+  beam-migrate = doJailbreak super.beam-migrate;
+
+  # dontcheck: Even though pg_ctl command is provided during tests it doesn't work.
+  beam-postgres = doJailbreak (dontCheck (overrideCabal super.beam-postgres (drv: {
     testToolDepends = (drv.testToolDepends or []) ++ [pkgs.postgresql];
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-common.nix:1038]: I think it's time to simplify this :) How about:

```suggestion
  beam-postgres = overrideCabal super.beam-postgres (drv: {
    jailbreak = true;
    doCheck = false;
    testToolDepends = (drv.testToolDepends or []) ++ [pkgs.postgresql];
```

**Review by sternenseemann — COMMENTED** [review]
These all build now, thanks!

It'd be nice if you could open issues about the version bounds upstream and link the issues in comments next to the respective overrides, so it's easy to see when they can removed.

**Comment by ZzAntares** [comment]
It seems is no longer necessary to report this upstream, [they have already loosen version bounds on aeson](https://github.com/haskell-beam/beam/commit/d73cc50ca41b44b40d086356298fa5150fd040c1) it should be available in the next beam's release.

Let me know if we should still attach a _"Needed till next beam's release"_ note though.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 2]
```diff
@@ -1027,10 +1027,18 @@ self: super: {
...
+  # Aeson dependency version bound not satifised but it builds fine without it
+  beam-migrate = doJailbreak super.beam-migrate;
+
+  # dontcheck: Even though pg_ctl command is provided during tests it doesn't work.
   beam-postgres = overrideCabal super.beam-postgres (drv: {
+    jailbreak = true;
+    doCheck = false;
     testToolDepends = (drv.testToolDepends or []) ++ [pkgs.postgresql];
-    });
+    })));
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-common.nix:1041]: These parens are now too much, I forgot to include that line in my suggestion.

```suggestion
    });
```

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 3]
```diff
@@ -331,7 +331,7 @@ self: super: {
   pipes-websockets = dontCheck super.pipes-websockets;
   posix-pty = dontCheck super.posix-pty; # https://github.com/merijn/posix-pty/issues/12
   postgresql-binary = dontCheck super.postgresql-binary; # needs a running postgresql server
-  postgresql-simple-migration = dontCheck super.postgresql-simple-migration;
+  postgresql-simple-migration = doJailbreak (dontCheck super.postgresql-simple-migration);
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-common.nix:334]: Is there an issue / fix for the version constraints for `postgresql-simple-migration` upstream?

**Comment by sternenseemann** [comment]
> Let me know if we should still attach a "Needed till next beam's release" note though.

Grouping them together and adding a comment like “beam overrides are needed until the next release (current is 0.9.0.0)” would be great, yeah!

**Pull request closed without merging**